src/protocol/helpers.ts

```diff
@@ -0,0 +1,33 @@
+function formatValue(value: any): string {
+    if (Array.isArray(value)) {
+        return '[' + value.map((v) => formatValue(v)).join(',') + ']'
+    } else if (!isNaN(value)) {
+        return value.toString()
+    } else {
+        return `'${value}'`
+    }
+
+}
+
+/**
+ * Form a log-friendly string representation of a message
+ * @param className -- name of the message class
+ * @param addEndingDots -- whether to add "..." at the end of the body, signalling that some values are not shown
+ * @param args -- pairs of (key, value) to print out, .e.g, ['streamId', '7wa7APtlTq6EC5iTCBy6dw'] => streamId='7wa7APtlTq6EC5iTCBy6dw'
+ */
+export function formLogFriendlyString(className: string, addEndingDots: boolean, ...args: any[]) {
```

Comment:
the `key1, value2, key2, value2)` args seems… not very ergonomic, you've got a mapping of keys and values… which sounds a lot like an object. And maybe the source object itself could be that object… read the values directly off that, based on a set of given keys? 

The "classname" is also usually readable from `object.constructor.name`.

`addEndingDots` could be derived from whether you printed all the keys.

Also perhaps could utilise `util.format` or `util.inspect` e.g. something like:
```js
const util = require('util')
function formLogFriendlyString(obj, keys) {
    const keySet = new Set(keys)
    const addEndingDots = Object.keys(obj).every((key) => keySet.has(key))
    const body = !keys ? obj : keys.reduce((o, key) => {
        o[key] = obj[key]
        return o
    }, {})

    return `${obj.constructor.name} ${util.inspect(body, {
        colors: process?.stdout?.isTTY,
    })}${addEndingDots ? ' ... ' : ''}`
}

class User {
    constructor(opts) {
        Object.assign(this, opts)
    }
}

console.log(formLogFriendlyString(new User({
    id: 32,
    name: 'test',
    age: 52,
}), ['id', 'name']))

```

![image](https://user-images.githubusercontent.com/43438/111668588-484d7700-87ec-11eb-963d-e1e9a32805f1.png)

Also could look into implementing this as a reusable custom inspect property, this is kinda what it's for: https://nodejs.org/api/util.html#util_custom_inspection_functions_on_objects

A custom inspect which calls a custom `toJSON` method could be the thing that does the `streamId: getStreamId()` thing:

```js
class StreamMessage {
    constructor(id) {
        this._id = id
    }
    getId() {
        return this._id
    }
    toJSON() {
        return { id: this.getId()
        }
    }
}
```

```
const s = new StreamMessage('test-id')
> JSON.stringify(s)
'{"id":"test-id"}'
> s.toJSON()
{ id: 'test-id' }
```

```js
class StreamMessage {
    constructor(id) {
        this._id = id
    }
    getId() {
        return this._id
    }
    toJSON() {
        return {
            id: this.getId()
        }
    }
    [util.inspect.custom](depth, options) {
        return `${options.stylize(this.constructor.name, 'special')} ${util.inspect(this.toJSON(), options)}`
    }
}

const s = new StreamMessage('test-id')
console.log('console.log(s)', s)
console.log('console.log(s.toJSON())', s.toJSON())
console.log('console.log(util.inspect(s))', util.inspect(s, {
    colors: process?.stdout?.isTTY,
}))
```
![image](https://user-images.githubusercontent.com/43438/111671310-36210800-87ef-11eb-8579-0e183c0b4751.png)

----

Maybe could extend the list of valid keys to support nested/chained keys like `msg.streamMessage.messageId` and use https://lodash.com/docs/4.17.15#get to grab them.